pkg/cstor/pv_operation.go

```diff
@@ -58,7 +59,59 @@ func (p *Plugin) updateVolCASInfo(data []byte, volumeID string) error {
 	return nil
 }
 
-func (p *Plugin) restoreVolumeFromCloud(vol *Volume) error {
+// restoreVolumeFromCloud restore remote snapshot for the given volume
+// Note: cstor snapshots are incremental in nature, so restore will be executed
+// from base snapshot to incremental snapshot 'vol.backupName' if p.restoreAllSnapshots is set
+// else restore will be performed for the given backup only.
+func (p *Plugin) restoreVolumeFromCloud(vol *Volume, targetBackupName string) error {
+	var (
+		snapshotList []string
+		err          error
+	)
+
+	if p.restoreAllSnapshots {
+		// We are restoring from base backup to targeted Backup
+		snapshotList, err = p.cl.GetSnapListFromCloud(vol.snapshotTag, p.getScheduleName(targetBackupName))
+		if err != nil {
+			return err
+		}
+	} else {
+		// We are restoring only given backup
+		snapshotList = []string{targetBackupName}
+	}
+
+	if len(snapshotList) == 0 {
```

Comment:
after below check for ` 	if !contains(snapshotList, targetBackupName) `, we don't need this one.

Reply:
ok